runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -921,20 +925,499 @@ static void VMwrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register
+ * arg3 pointer to value being written
+ *
+ * jitReportStaticFieldRead
+ * arg1 pointer to static data block
+ *
+ * jitReportStaticFieldWrite
+ * arg1 pointer to static data block
+ * arg2 pointer to value being written
+ *
+ */
+void generateReportFieldAccessOutlinedInstructions(TR::Node *node, TR::LabelSymbol *endLabel, TR::Register *dataBlockReg, bool isWrite, TR::RegisterDependencyConditions *deps, TR::CodeGenerator *cg, TR::Register *sideEffectRegister, TR::Register *valueReg)
+   {
+   
+   bool is64Bit = TR::Compiler->target.is64Bit();
+   bool isInstanceField = node->getSymbolReference()->getSymbol()->getKind() != TR::Symbol::IsStatic;
+   bool isResolved = !node->getSymbolReference()->isUnresolved();
+   J9Method *owningMethod = (J9Method *)node->getOwningMethod();
+
+   TR_RuntimeHelper helperIndex = isWrite ? (isInstanceField ? TR_jitReportInstanceFieldWrite: TR_jitReportStaticFieldWrite):
+                                            (isInstanceField ? TR_jitReportInstanceFieldRead: TR_jitReportStaticFieldRead);
+
+   TR::Linkage *linkage = cg->getLinkage(runtimeHelperLinkage(helperIndex));
+   auto linkageProperties = linkage->getProperties();
+   TR::Register *valueReferenceReg = NULL;
+   TR::MemoryReference *valueMR = NULL;
+
+   TR_BackingStore * location = NULL;
+   TR::Compilation * comp = cg->comp();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *) (cg->fe());
+   
+
+   /*
+    * For reporting field write, reference to the valueNode is needed so we need to store
+    * the value on to a stack location first and pass the stack location address as an arguement
+    * to the VM helper
+    */
+   if (isWrite)
+      {
+      bool isUnsigned = node->getOpCode().isUnsigned();
+      TR::DataType dt = node->getDataType();
+      int32_t elementSize = dt == TR::Address ? TR::Compiler->om.sizeofReferenceField() : TR::Symbol::convertTypeToSize(dt);
+      TR::InstOpCode::Mnemonic storeOp = getLoadOrStoreFromDataType(cg, dt, elementSize, isUnsigned, false);
+      location = cg->allocateSpill(node->getSize(), false, NULL);
+      valueMR = new (cg->trHeapMemory()) TR::MemoryReference(node, location->getSymbolReference(), node->getSize(), cg);
+
+      if (!valueReg->getRegisterPair())
+         {
+         // 64Bit Long
+         generateMemSrc1Instruction(cg, storeOp, node, valueMR, valueReg);
+         }
+      else
+         {
+         //32 Bit long
+         generateMemSrc1Instruction(cg, TR::InstOpCode::stw, node, new (cg->trHeapMemory()) TR::MemoryReference(node, location->getSymbolReference(), 4, cg), valueReg->getLowOrder());
+         generateMemSrc1Instruction(cg, TR::InstOpCode::stw, node, new (cg->trHeapMemory()) TR::MemoryReference(node, location->getSymbolReference() + 4, 4, cg), valueReg->getHighOrder());
+         }
+      deps->addPreCondition(valueReg, TR::RealRegister::NoReg);
+      deps->addPostCondition(valueReg, TR::RealRegister::NoReg);
+
+      //store the stack location into a register
```

Comment:
To stay consistent with the convention in _most_ of the codebase, can we leave a space after the comment? Ex: 
```
// store the stack location into a register
```

Similarly everywhere else.